### PR TITLE
ALGO-1061 update api client

### PR DESCRIPTION
**Pull request by zeryx**
This PR modifies the API client interface behaviour and some test cases to ensure that we're able to provide support for new endpoints and functionality generated from the openapi spec, in the python client.

Relies on https://github.com/algorithmiaio/algorithmia-api-client/pull/10, which hasn't been deployed to pypitest or pypi. To test, simply git clone the above branch, and `python setup.py bdist_wheel` followed by a `pip install -U --force-reinstall dist/*`
You should now have an adequate version of algorithmia-api-client installed to run `pytest` on this branch.

### CODE REVIEW

File: Algorithmia/algorithm.py [hunk 1]
```diff
@@ -41,31 +41,23 @@ def set_options(self, timeout=300, stdout=False, output=OutputType.default, **qu
     def create(self, details={}, settings={}, version_info={}):
         detailsObj = Details(**details)
         settingsObj = SettingsMandatory(**settings)
-        createRequestVersionInfoObj = CreateRequestVersionInfo(**version_info)
+        createRequestVersionInfoObj = VersionInfo(**version_info)
         create_parameters = {"name": self.algoname, "details": detailsObj, "settings": settingsObj, "version_info": createRequestVersionInfoObj}
         create_request = CreateRequest(**create_parameters)
-        try:
```
Comment (zeryx) [inline Algorithmia/algorithm.py:47]: This try/catch was removed as the intended purpose of RaiseAlgoApiErrors was to handle algorithm request errors. Those have been removed as they swallow critical stack traces and other important pieces of information

File: Algorithmia/algorithm.py [hunk 2]
```diff
@@ -74,53 +66,36 @@ def publish(self, details={}, settings={}, version_info={}):
...
-            else:
-                api_response = self.client.manageApi.get_algorithm_builds(self.username, self.algoname, limit=limit)
-            return api_response
-        except ApiException as e:
-            error_message = json.loads(e.body)
-            raise raiseAlgoApiError(error_message)
+        if marker is not None:
+            api_response = self.client.buildsApi.get_algorithm_builds(self.username, self.algoname, limit=limit, marker=marker)
+        else:
+            api_response = self.client.buildsApi.get_algorithm_builds(self.username, self.algoname, limit=limit)
```
Comment (kennydaniel) [inline Algorithmia/algorithm.py]: Is this if-else necessary? Seems like same default argument?

File: Algorithmia/client.py [hunk 3]
```diff
@@ -53,9 +53,11 @@ def __init__(self, apiKey = None, apiAddress = None, caCert = None):
         if not config:
             config = Configuration()
         config.api_key['Authorization'] = self.apiKey
-        config.host = "{}/v1".format(self.apiAddress)
-        self.manageApi = DefaultApi(ApiClient(config))
-        
+        config.host = "{}".format(self.apiAddress)
+        self.buildsApi = BuildsApi(ApiClient(config))
+        self.algorithmApi = AlgorithmsApi(ApiClient(config))
+        self.scmApi = ScmApi(ApiClient(config))
```
Comment (kennydaniel) [inline Algorithmia/client.py]: deleteable?

Comment (zeryx) [reply Algorithmia/client.py]: yes!

File: Algorithmia/algorithm.py (hunk 2 above)
Comment (zeryx) [reply Algorithmia/algorithm.py]: correct it can be removed!

**Review by kennydaniel — DISMISSED** [review]

**Pull request closed without merging**